### ISSUE TITLE
Present result-result comparison neutrally, 'baseline' & 'contender' is confusing

### ISSUE THREAD

**Issue by jgehrcke** [conbench/conbench#1369]
From https://github.com/conbench/conbench/pull/1336#discussion_r1245036329:

> I was also thinking that maybe what matters is "old" vs. "new".

> when comparing just two results I think I like when the presentation is almost neutral as in A vs. B without assigning specific properties / bias.

Antoine's feedback when looking at a result-result compare page:

> contender commit has an earlier date than baseline commit?

**Comment by austin3dickey** [comment]
I agree that we can be more simple with this language in the UI, and probably API.

When it comes to alerts, are we still open to using "baseline" and "contender"? The alerting workflow is a special case where we can potentially be more precise.

**Comment by alistaire47** [comment]
Yeah, I think in situations where you're getting a comparison via a baseline proposed by Conbench the existing language is good.

I even think we could keep the language entirely as long as we warn when the timestamps are backwards; this might occasionally be intentional, but it's usually by mistake. But I'm open to changing the terminology if it's helpful from a UX perspective

**Comment by jgehrcke** [comment]
> are we still open to using "baseline" and "contender"? 

Oh, I feel like this is a great question! And maybe we want to change something about that in the future. For example, I like "reference distribution".

I want to clarify that in this ticket I was thinking exclusively about the UI view at `/compare/benchmark-results/a...b`; I think on that view we do not need the b-word and the c-word :).

**Comment by austin3dickey** [comment]
>And maybe we want to change something about that in the future. For example, I like "reference distribution".

Yes, we'll probably want to re-examine the "baseline" word in the context of the lookback z-score analysis.

---

Thinking about language more, "old" vs "new" might be misleading in some cases. What if the "baseline" results are actually a re-run of benchmarks on an old commit, with a newer start timestamp than the "contender" start timestamp? Or, if "new" vs "old" refers to commit time instead of benchmark start time, what about git graphs where the "baseline" commit is newer than the "contender" commit because we're comparing to a distribution of results on a different branch?

**Comment by jgehrcke** [comment]
> Thinking about language more, "old" vs "new" might be misleading in some cases.

:+1:  let's just make this as neutral as possible, not assigning special meaning to any of the two results.

The UI view basically shows two columns, and here we wonder about the column headings. We can simply have them be `result <ID1>` and `result <ID2>`, order as defined by the two IDs in the URL.

**Comment by austin3dickey** [comment]
That makes sense to me!

**Comment by alistaire47** [comment]
As a user, I don't care what they're called; I'm still going to be confused if the left-hand result has a commit time after the right-hand one.

**Comment by jgehrcke** [comment]
> As a user, I don't care what they're called;

I don't quite believe that :P Jim and Chantal!

> I'm still going to be confused if the left-hand result has a commit time after the right-hand one.

OK, I respect that :) Still, I think it will be a bit of an improvement to remove the more heavy-weight terms for now.

In this case here I think the order of IDs in the URL should have a meaning. I mean that we should not automatically re-order. With that we are not deciding anything now, but retain flexibility to decide at the time of URL generation. Maybe not the strongest argument, but https://github.com/conbench/conbench/compare/6ab7ff096c2dc31a49fb9b0078935858c11a7c59...40a78165e02e781edc91d795bba3cbd34010e4aa yields a different view than https://github.com/conbench/conbench/compare/40a78165e02e781edc91d795bba3cbd34010e4aa...6ab7ff096c2dc31a49fb9b0078935858c11a7c59 (there is no auto-magic normalization "you probably meant ... and I did that for you").

**Comment by jgehrcke** [comment]
Addressed for now via #1375.

**Comment by alistaire47** [comment]
> In this case here I think the order of IDs in the URL should have a meaning. I mean that we should not automatically re-order. With that we are not deciding anything now, but retain flexibility to decide at the time of URL generation. Maybe not the strongest argument, but [6ab7ff0...40a7816](https://github.com/conbench/conbench/compare/6ab7ff096c2dc31a49fb9b0078935858c11a7c59...40a78165e02e781edc91d795bba3cbd34010e4aa) yields a different view than [40a7816...6ab7ff0](https://github.com/conbench/conbench/compare/40a78165e02e781edc91d795bba3cbd34010e4aa...6ab7ff096c2dc31a49fb9b0078935858c11a7c59) (there is no auto-magic normalization "you probably meant ... and I did that for you").

Agreed, there are (rare) good reasons for timestamps being flipped, so I don't think normalization is a good move. But I do think a warning (maybe with a link to the flipped version) would be helpful.

**Comment by austin3dickey** [comment]
@pitrou brought up that after this change, in the current page, it's hard to identify which column corresponds to which plotted `X`:
<img width="1358" alt="image" src="https://github.com/conbench/conbench/assets/16600275/1ba97c8a-940d-479a-89b3-d5864f6c990f">
Even if we want to remain neutral in terms of assuming which result is the "baseline" etc, I think we should at least label the columns with a way to identify which plot point they correspond to.
